### PR TITLE
[SDEV-1812] add ruler to fastem toolbar

### PR DESCRIPTION
**Pull request by nandishjpatel**

### CODE REVIEW

File: src/odemis/gui/cont/tabs/fastem_main_tab.py [hunk 1]
```diff
@@ -60,7 +67,13 @@ def __init__(self, name, button, panel, main_frame, main_data):
           Handles FastEMOverviewTab and FastEMAcquisitionTab.
 
         """
-        tab_data = guimod.FastEMMainTabGUIData(main_data)
+        tab_data = FastEMMainTabGUIData(main_data)
+        tab_data.tool.choices = {
+            TOOL_NONE,
+            TOOL_ACT_ZOOM_FIT,
+            TOOL_RULER,
+            TOOL_LINE,
```
Comment (tepals) [inline src/odemis/gui/cont/tabs/fastem_main_tab.py:75]: I tried out your code and the line tool does not work. Do you need the line tool?

Comment (nandishjpatel) [reply src/odemis/gui/cont/tabs/fastem_main_tab.py:75]: The line tool is to move the scale and extend the edges of the scale. Can you give it a try?

Comment (tepals) [reply src/odemis/gui/cont/tabs/fastem_main_tab.py:75]: Do you mean the move the edges of the ruler? That indeed works. I do sometimes get the following error:
```
09:09:11 ERROR   log            : Subscriber <odemis.util.weak.WeakMethodBound object at 0x7fb0207de700> raised exception when receiving value 4
Traceback (most recent call last):
  File "/home/pals/development/odemis/src/odemis/model/_vattributes.py", line 96, in notify
    l(v)
  File "/home/pals/development/odemis/src/odemis/util/weak.py", line 36, in __call__
    return self.f(ins, *arg, **kwargs)
  File "/home/pals/development/odemis/src/odemis/gui/cont/tabs/fastem_main_tab.py", line 181, in _on_tool_va
    self.vp.canvas.disable_drag()
  File "/home/pals/development/odemis/src/odemis/gui/comp/canvas.py", line 1447, in disable_drag
    self.abilities.remove(CAN_DRAG)
KeyError: 1
```

Comment (nandishjpatel) [reply src/odemis/gui/cont/tabs/fastem_main_tab.py:75]: My bad, yes to move the edges of the ruler. Nice catch I will fix this.

File: src/odemis/gui/cont/tabs/fastem_main_tab.py [hunk 2]
```diff
@@ -154,6 +168,20 @@ def __init__(self, name, button, panel, main_frame, main_data):
...
+        self.tb.add_tool(TOOL_ACT_ZOOM_FIT, self.view_controller.fitViewToContent)
+        self.tb.add_tool(TOOL_RULER, self.tab_data_model.tool)
+        self.tb.add_tool(TOOL_LINE, self.tab_data_model.tool)
+
+    def _on_tool_va(self, selected_tool):
+        if selected_tool == TOOL_NONE:
+            self.vp.canvas.enable_drag()
+            return
+        if CAN_DRAG in self.vp.canvas.abilities:
+            self.vp.canvas.disable_drag()
```
Comment (pieleric) [inline src/odemis/gui/cont/tabs/fastem_main_tab.py:183]: I don't think you need to change the drag capability. Normally, when the gadget overlay is active, it gets first the mouse events, and never pass them to the canvas. So the drag doesn't even start.

File: src/odemis/gui/cont/tabs/fastem_main_tab.py (hunk 1 above)
Comment (nandishjpatel) [reply src/odemis/gui/cont/tabs/fastem_main_tab.py]: I fixed it!

File: src/odemis/gui/comp/overlay/stage_point_select.py [hunk 3]
```diff
@@ -29,27 +29,13 @@ class StagePointSelectOverlay(WorldOverlay):
...
-        if self.active:
+        if self.active.value:
             v_pos = evt.Position
             p_pos = self.cnvs.view_to_phys(v_pos, self.cnvs.get_half_buffer_size())
             # directly move the stage to the selected physical position
             self.cnvs.view.moveStageTo(p_pos)
         else:
             WorldOverlay.on_dbl_click(self, evt)
 
-    def on_left_down(self, evt):
```
Comment (nandishjpatel) [inline src/odemis/gui/comp/overlay/stage_point_select.py:40]: @K4rishma this class is the base class for cryo feature overlay. Do you know what will happen if `on_left_down` and `on_left_up` are removed?

Comment (nandishjpatel) [reply src/odemis/gui/comp/overlay/stage_point_select.py:40]: I followed the software release procedure for METEOR and it seems to work fine with this change

**Review by tepals — APPROVED** [review]

**Pull request merged**